### ISSUE TITLE
Inquiring about Optimal Batch Size Adjustment Strategy for Different Hardware Configurations

### ISSUE THREAD

**Issue by Sun2018421** [Liuhong99/Sophia#54]
Hi authors,

I'm currently running the training script on 4x A100 80GB GPUs using the following command:

bash
```
torchrun --standalone --nproc_per_node=4 \
      train_sophiag_mini.py \
      config/train_gpt2_medium_sophiag_mini.py \
      --batch_size=6 \
      --gradient_accumulation_steps=10
```

However, the performance doesn't show the expected advantages compared to AdamW baseline results. 
Noticing your experiments used 8x A100 40GB configurations, I would like to ask:
Should I increase the per-GPU batch_size given that my A100s have larger memory (80GB vs 40GB)?

Are there recommended adjustment strategies when scaling down from 8-GPU to 4-GPU setups (e.g., proportional scaling rules for batch size/gradient accumulation steps)?
Would you suggest other hyperparameter modifications (like learning rate) to compensate for the hardware difference?
My current total batch size is 6 * 4 * 10 = 240 (batch_size * GPUs * gradient_accumulation). Could you advise whether this needs adjustment to better utilize my hardware configuration?

Thank you for your time and guidance. I would greatly appreciate any suggestions.

**Comment by Sun2018421** [comment]
My results are as follow, <img width="1474" alt="Image" src="https://github.com/user-attachments/assets/9b85652b-8c4c-4301-9e75-11f11dd5f648" />
I've reviewed this issue https://github.com/Liuhong99/Sophia/issues/46. My configuration matches the one described, but it doesn't seem to resolve my problem.

**Comment by Sun2018421** [comment]
i have revised my batch_size and grad_accum, and the problem may be addressed
